geo_tag/geo_tag.py

```diff
@@ -0,0 +1,200 @@
+import json
+import os
+import pickle
+
+from geopy.geocoders import Nominatim
+
+class TwitterJSONTagger():
+    def __init__(self):
+        # data is the tweet json which we add geo_tag.
+        self.data = dict()
+        # cache is the hashmap { (city, state), other geo info}
+        self.cache = dict()
+        # coord is the (longitude, latitude)
+        self.coord = tuple()
+        # self.abbrev_us_state is the relation between the state's full name and abbrev.
+        with open('us_state_abbrev.json', 'r') as inf:
+            self.abbrev_us_state = eval(inf.read())
+
+    def init_geo_cache(self, city_json):
```

Comment:
no need to take `city_json` as an argument. geotagger won't work without `city_json`.

Reply:
OK, so read it in the __init__ just like us_state_abbrev?